File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    
    priority_queue<int, vector<int>, greater<int>> pq;
    for (auto& i:scoville) pq.push(i);
    
    while (true){
        int first = pq.top();
        if (first >= K) break;
        pq.pop();
        if (!pq.empty()){
            int second = pq.top();
            pq.pop();
            pq.push(first+second*2);
            answer++;
        }
        else return -1;         
    }
    return answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    
    while scoville:
        first = heapq.heappop(scoville)
    
        if first >= K: break
        
        if scoville:
            second = heapq.heappop(scoville)
            heapq.heappush(scoville, first + second * 2)
        else:
            return -1
        
        answer += 1
                
    return answer